core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -141,6 +144,28 @@ class CityConstructions {
         return result
     }
 
+    fun getProductionMarkup(ruleset: Ruleset): FormattedLine {
+        val currentConstructionSnapshot = currentConstructionFromQueue
+        if (currentConstructionSnapshot.isEmpty()) return FormattedLine()
+        val category = when {
+            ruleset.buildings[currentConstructionSnapshot]
```

Comment:
Cannot be null - current construction from queue is checked upon GameInfo.setTransients() and removed if doesn't fit the ruleset. Change to !!.let{}

Reply:
The `.isEmpty()` on line 149 is there because `getProductionForTileInfo` did the same - but the result would be the same with the line removed.
`ruleset.buildings[currentConstructionSnapshot]` can be null if city is constructing a unit? Thing is, I only need to safely determine a civilopedia category with silent fallback to just not linking when things don't go as expected - or so I thought. The need to differentiate buildings from wonders is a bit painful and I pondered alternate approaches for a while but didn't come up with something really convincing.

Reply:
That's another edge case solved by our miraculous getCurrentConstruction() ;)